### PR TITLE
Fix macOS launcher app to allow release and preview versions.

### LINKED ISSUE #7074: PowerShell v6.1.0-preview.3 on macOS - PowerShell.app does not start PowerShell
on macOS, the PowerShell.app/Contents/MacOS/PowerShell.sh script does not invoke pwsh-preview; it incorrectly tries to invoke pwsh.

### PR DESCRIPTION
**Pull request by thezim**
## PR Summary

Fixes #7074. This PR allows concurrent versions of the macOS launcher app to exist. The application name will be either `PowerShell` or `PowerShell-preview`. It is now dynamically built and removed post `fpm` to avoid post build conflicts, see #5262 .

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [X] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [X] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: tools/packaging/packaging.psm1 [hunk 1]
```diff
@@ -1175,38 +1176,33 @@ function New-MacOSLauncher
...
-        defaults write $plist CFBundleIconFile $iconfilebase
-    }
-
-    # Convert to XML plist, needed because defaults native
-    # app auto converts it to binary format when it modify
-    # the plist file.
-    Start-NativeExecution {
-        plutil -convert xml1 $plist
-    }
+    $plistcontent = $packagingStrings.MacOSLauncherPlistTemplate -f $packageId, $Version, $iconfilebase
```
Comment (TravisEz13) [inline tools/packaging/packaging.psm1]: Did you forget to include this change to the strings?

**Review by TravisEz13 — CHANGES_REQUESTED** [review]
see comment

File: tools/packaging/packaging.psm1 (hunk 1 above)
Comment (thezim) [reply tools/packaging/packaging.psm1]: This has been corrected.

**Review by TravisEz13 — APPROVED** [review]
restarted travis-ci due to a spelling error introduced by another change

**Pull request merged**